### ISSUE TITLE
Seed all random actions

### ISSUE THREAD

**Issue by gavincmartin** [autognc/jigsaw#9]
We currently use the standard library's `random` module, `numpy.random` and random functionality with `sklearn`. We should find a way to set a single seed used for all random actions.

**Comment by gavincmartin** [comment]
From `cli.py`:
```
how_many = user_input(
            message='How many images of set "{}" would you like to use? (?/{})'
            .format(set_name, len(set_data)),
            validator=IntegerValidator)
        n = int(how_many)
        sets_to_join.append(set_data.sample(n, replace=False, random_state=42))
```
This is still separate from the seed used in `write_dataset.py` (partially because this is a pandas function that can use a numpy random state).

**Comment by gavincmartin** [comment]
Fixed with #15
